core/app/[locale]/(default)/(faceted)/search/page.tsx

```diff
@@ -244,7 +245,10 @@ export async function generateMetadata() {
   };
 }
 
-export default async function Search(props: Props) {
+export default async function Search(pageProps: Props) {
+  const searchParams = PLazy.from(() => pageProps.searchParams);
+  const props = { searchParams };
```

Comment:
If we're seeing problems with `searchParams` similar to `cookies` I think that we might need a different approach here...

Having to use a lazy promise here feels wrong to me. In particular, `Search` should be dynamic always.

Reply:
I agree with @migueloller on this. Search is being rendered dynamically though.

Reply:
>Search should be dynamic always.
Wait, with this change, search is still dynamic. This change just defers the `searchParams` execution to component that await the function.

But I agree with the general sentiment though. This is a landmine. I don't want developers to think about this quirk while developing Catalyst. I'll close